models/DataObject/Data/ObjectMetadata.php

```diff
@@ -54,8 +55,26 @@ class ObjectMetadata extends Model\AbstractModel implements DataObject\OwnerAwar
     public function __construct($fieldname, $columns = [], $object = null)
     {
         $this->fieldname = $fieldname;
-        $this->object = $object;
         $this->columns = $columns;
+        $this->setObject($object);
```

Comment:
saving an object with lot of relations seems still quite expensive, event if the relations didn't change.

Reply:
the dirty map gets reset when loading is finished, so there isn't an issue

Reply:
ok, saving is expensive because getter is called in Object\Concrete `update()` method, see also https://github.com/pimcore/pimcore/issues/4154 
has nothing to do with this PR, there everything is good.